### ISSUE TITLE
Security reports with path-to-regexp version

### ISSUE THREAD

**Issue by javierjulio** [firebase/superstatic#478]
We have two [Dependabot security reports regarding the path-to-regexp version](https://github.com/advisories/GHSA-9wv6-86v2-598j) and since we use firebase-tools, I believe it's coming from this package. This Dependabot PR https://github.com/firebase/superstatic/pull/476 here would resolve it since the router package also depends on a vulnerable path-to-regexp version so both need to be updated. Worth noting that the router package is a major version update.

**Comment by baraknaveh** [comment]
Same here, would be great if the team can give #476 a little love...

**Comment by cam-at-tactiq** [comment]
This is affecting us too.

**Comment by mgm793** [comment]
Any update here?

**Comment by mgm793** [comment]
I created a PR to fix it: https://github.com/firebase/superstatic/pull/482

**Comment by IchordeDionysos** [comment]
This is now fixed

**Comment by moekify** [comment]
@IchordeDionysos This is not fixed yet, CVE-2024-45296 is still open https://github.com/pillarjs/path-to-regexp/security/advisories/GHSA-9wv6-86v2-598j in this repository given that path-to-regexp is still on `v1.8.0`

**Comment by IchordeDionysos** [comment]
True, the issue here is fixed though downstream consumers are not locked into a vulnerable version as the lockfile of superstatic shouldn't be used.

Unless you use superstatic directly (which is likely the minority of the cases) you are not blocked.

**Comment by moekify** [comment]
@IchordeDionysos Yes, but it means we need to monkeypatch and override a version. It would be preferable to deal with security vulnerabilities at the source. Most people probably come here via `firebase-tools`

**Comment by joehan** [comment]
This is fixed after the recent package-lock.json regen, but https://github.com/firebase/superstatic/pull/488 will update package.json to match.

**Comment by moekify** [comment]
Thank you're @joehan!

**Comment by moekify** [comment]
@joehan Are you also tagging a new release `9.2.0` so the changes can be picked up upstream in `firebase-tools`? :)

**Comment by joehan** [comment]
@moekify This is gonna go out in a patch release soon (we try to follow semver accurately, and there are no minor changes queued up). We will pick this up in firebase-tools by updating the package.json there
